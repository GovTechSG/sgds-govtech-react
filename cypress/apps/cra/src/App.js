import 'bootstrap-icons/font/bootstrap-icons.css';
import '@govtechsg/sgds/sass/sgds.scss';
import '@govtechsg/sgds-masthead/dist/sgds-masthead/sgds-masthead.css';

import {
  FileUploadCom,
  MastheadCom,
  TooltipCom,
  NavbarCom,
  StepperCom,
  AccordionCom,
  AlertCom,
  BadgeCom,
  BreadcrumbCom,
  CardCom,
  QtyToggleCom,
  SelectableCardCom,
  DropdownCom,
  ValidationCom,
  InputGroupCom,
  TableCom,
  ToastCom,
  PaginationCom,
  SideNavCom,
  ProgressBarCom,
  DatePickerCom,
  ComboboxCom,
  ModalCom,
  TabsCom
} from './components';

function App() {
  return (
    <div className="container">
      <MastheadCom />
      <FileUploadCom />
      <TooltipCom />
      <NavbarCom />
      <StepperCom />
      <AccordionCom />
      <AlertCom />
      <BadgeCom/>
      <BreadcrumbCom/>
      <CardCom/>
      <QtyToggleCom/>
      <SelectableCardCom/>
      <DropdownCom/>
      <ValidationCom/>
      <InputGroupCom/>
      <TableCom/>
      <ToastCom/> 
      <PaginationCom/>
      <SideNavCom/>
      <ProgressBarCom/> 
      <DatePickerCom/>
      <ComboboxCom/>
      <ModalCom/>
      <TabsCom/>
    </div>
  );
}

export default App;
