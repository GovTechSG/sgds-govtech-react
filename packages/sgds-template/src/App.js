import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import { NavbarTemp } from "./components/NavbarTemp";
import { SinglePageFormTemp } from "./components/SinglePageFormTemp";
import { MultiPageFormTemp } from "./components/MultiPageFormTemp";
import { CollapsibleFormTemp } from "./components/CollapsibleFormTemp";
import "@govtechsg/sgds-govtech/sgds/sgds.css";

function App() {
  return (
    <Router>
      <div className='mb-3'>
        <NavbarTemp />
      </div>
      <div className='mb-3'>
        <Routes>
          <Route path='/' element={<SinglePageFormTemp />} />
          <Route path='/multi' element={<MultiPageFormTemp />} />
          <Route path='/collapsible' element={<CollapsibleFormTemp />} />
        </Routes>
      </div>
    </Router>
  );
}

export default App;
