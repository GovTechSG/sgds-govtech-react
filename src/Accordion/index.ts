export { default as Accordion } from './Accordion';
export type { AccordionProps } from './Accordion';

export { default as AccordionContext } from './AccordionContext';
export { default as AccordionCollapse } from './AccordionCollapse';
export type { AccordionCollapseProps } from './AccordionCollapse';

export {
  default as AccordionButton,
  useAccordionButton,
} from './AccordionButton';
export type { AccordionButtonProps } from './AccordionButton';

export { default as AccordionItem } from './AccordionItem';
export type { AccordionItemProps } from './AccordionItem';
export { default as AccordionHeader } from './AccordionHeader';
export type { AccordionHeaderProps } from './AccordionHeader';
export { default as AccordionBody } from './AccordionBody';
export type { AccordionBodyProps } from './AccordionBody';
