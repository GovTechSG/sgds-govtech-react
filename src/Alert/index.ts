export { default as Alert } from './Alert';
export type { AlertProps } from './Alert';

export { default as AlertLink } from './AlertLink';
// export type { AlertLinkProps } from './AlertLink';

export { default as AlertHeading } from './AlertHeading';
