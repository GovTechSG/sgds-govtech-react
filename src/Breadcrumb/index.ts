export { default as Breadcrumb } from './Breadcrumb';
export type { BreadcrumbProps } from './Breadcrumb';

export { default as BreadcrumbItem } from './BreadcrumbItem';
export type { BreadcrumbItemProps } from './BreadcrumbItem';