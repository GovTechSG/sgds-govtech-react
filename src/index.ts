export * from './Accordion'
// export { default as Accordion } from './Accordion/Accordion';
// export type { AccordionProps } from './Accordion/Accordion';

// export { default as AccordionContext } from './Accordion/AccordionContext';
// export { default as AccordionCollapse } from './Accordion/AccordionCollapse';
// export type { AccordionCollapseProps } from './Accordion/AccordionCollapse';

// export {
//   default as AccordionButton,
//   useAccordionButton,
// } from './Accordion/AccordionButton';
// export type { AccordionButtonProps } from './Accordion/AccordionButton';


export * from './Alert'
// export { default as Alert } from './Alert/Alert';
// export type { AlertProps } from './Alert/Alert';

// export { default as Anchor } from './Anchor';
// export type { AnchorProps } from './Anchor';
export * from './Badge'
// export { Badge } from './Badge/Badge';
// export type { BadgeProps } from './Badge/Badge';
export * from './Breadcrumb'
// export { default as Breadcrumb } from './Breadcrumb/Breadcrumb';
// export type { BreadcrumbProps } from './Breadcrumb/Breadcrumb';

// export { default as BreadcrumbItem } from './Breadcrumb/BreadcrumbItem';
// export type { BreadcrumbItemProps } from './Breadcrumb/BreadcrumbItem';

// export { default as Button } from './Button/Button';
// export type { ButtonProps } from './Button/Button';
export * from './Button'
export * from './ButtonGroup'
// export { default as ButtonGroup } from './ButtonGroup';
// export type { ButtonGroupProps } from './ButtonGroup';

// export { default as ButtonToolbar } from './ButtonToolbar';
// export type { ButtonToolbarProps } from './ButtonToolbar';
export * from './Card'
//   export { default as Card } from './Card/Card';
// export type { CardProps } from './Card/Card';
// export { default as SelectableCard } from './Card/SelectableCard'
// export type { SelectableCardProps } from './Card/SelectableCard'

// export { default as CardImg } from './Card/CardImg';
// export type { CardImgProps } from './Card/CardImg';

// export { default as CardGroup } from './Card/CardGroup';
// export { default as Carousel } from './Carousel';
// export type { CarouselProps } from './Carousel';

// export { default as CarouselItem } from './CarouselItem';
// export type { CarouselItemProps } from './CarouselItem';
export * from './CloseButton'
// export { default as CloseButton } from './CloseButton/CloseButton';
// export type { CloseButtonProps } from './CloseButton/CloseButton';
export * from './Col'
// export { default as Col } from '../Col';
// export type { ColProps } from '../Col';
export * from './CloseButton'
export * from './Collapse'
// export { default as Collapse } from './Collapse/Collapse';
// export type { CollapseProps } from './Collapse/Collapse';
export * from './Dropdown'
// export { default as Dropdown } from './Dropdown/Dropdown';
// export type { DropdownProps } from './Dropdown/Dropdown';

// export { default as DropdownButton } from './Dropdown/DropdownButton';
// export type { DropdownButtonProps } from './Dropdown/DropdownButton';
export * from './Fade'
// export { default as Fade } from './Fade/Fade';
// export type { FadeProps } from './Fade/Fade';
export * from './Form'
// export { default as Form } from './Form/Form';
// export type { FormProps } from './Form/Form';

// export { default as FormControl } from './Form/FormControl';
// export type { FormControlProps } from './Form/FormControl';

// export { default as FormCheck } from './Form/FormCheck';
// export type { FormCheckProps } from './Form/FormCheck';

// export { default as FormFloating } from './Form/FormFloating';
export * from './FloatingLabel'
// export { default as FloatingLabel } from '../FloatingLabel';
// export type { FloatingLabelProps } from '../FloatingLabel';

// export { default as FormGroup } from './Form/FormGroup';
// export type { FormGroupProps } from './Form/FormGroup';

// export { default as FormLabel } from './Form/FormLabel';
// export type { FormLabelProps } from './Form/FormLabel';

// export { default as FormText } from './Form/FormText';
// export type { FormTextProps } from './Form/FormText';

// export { default as FormSelect } from './Form/FormSelect';
// export type { FormSelectProps } from './Form/FormSelect';

// export { default as Container } from './Container';
// export type { ContainerProps } from './Container';

// export { default as Image } from './Image';
// export type { ImageProps } from './Image';

// export { default as Figure } from './Figure';
export * from './InputGroup'
// export { default as InputGroup } from './InputGroup/InputGroup';
// export type { InputGroupProps } from './InputGroup/InputGroup';

// export { default as ListGroup } from './ListGroup';
// export type { ListGroupProps } from './ListGroup';

// export { default as ListGroupItem } from './ListGroupItem';
// export type { ListGroupItemProps } from './ListGroupItem';
export * from './Modal'
// export { default as Modal } from './Modal/Modal';
// export type { ModalProps } from './Modal/Modal';

// export { default as ModalBody } from './Modal/ModalBody';
// export { default as ModalDialog } from './Modal/ModalDialog';
// export type { ModalDialogProps } from './Modal/ModalDialog';

// export { default as ModalFooter } from './Modal/ModalFooter';
// export { default as ModalTitle } from './Modal/ModalTitle';
export * from './Nav'
// export { default as Nav } from '../Nav/Nav';
// export type { NavProps } from '../Nav/Nav';

// // export { default as Navbar } from './Navbar';
// // export type { NavbarProps } from './Navbar';

// // export { default as NavbarBrand } from './NavbarBrand';
// // export type { NavbarBrandProps } from './NavbarBrand';

// // export { default as NavDropdown } from './NavDropdown';
// // export type { NavDropdownProps } from './NavDropdown';

// export { default as NavItem } from './Nav/NavItem';

// export { default as NavLink } from './Nav/NavLink';
// export type { NavLinkProps } from './Nav/NavLink';

// export { default as Offcanvas } from './Offcanvas';
// export type { OffcanvasProps } from './Offcanvas';
// export { default as OffcanvasHeader } from './OffcanvasHeader';
// export type { OffcanvasHeaderProps } from './OffcanvasHeader';
// export { default as OffcanvasTitle } from './OffcanvasTitle';
// export { default as OffcanvasBody } from './OffcanvasBody';
export * from './Overlay'
// export { default as Overlay } from './Overlay/Overlay';
// export type { OverlayProps } from './Overlay/Overlay';

// export { default as OverlayTrigger } from './Overlay/OverlayTrigger';
// export type { OverlayTriggerProps } from './Overlay/OverlayTrigger';
export * from './PaginationBase'
// export { default as PageItem } from './PaginationBase/PageItem';
// export type { PageItemProps } from './PaginationBase/PageItem';

// export { default as PaginationBase } from './PaginationBase/PaginationBase';
// export type { PaginationBaseProps } from './PaginationBase/PaginationBase';
export * from './Paginations'
// export { Paginations } from './Paginations/Paginations';
// export type { PaginationsProps } from './Paginations/Paginations';

// export { default as Placeholder } from './Placeholder';
// export type { PlaceholderProps } from './Placeholder';
// export { default as PlaceholderButton } from './PlaceholderButton';
// export type { PlaceholderButtonProps } from './PlaceholderButton';
export * from './Popover'
// export { default as Popover } from './Popover';
// export type { PopoverProps } from './Popover';

// export { default as PopoverHeader } from './PopoverHeader';
// export { default as PopoverBody } from './PopoverBody';
export * from './ProgressBar'
// export { default as ProgressBar } from './ProgressBar/ProgressBar';
// export type { ProgressBarProps } from './ProgressBar/ProgressBar';
export * from './QuantityToggle'
// export {default as QuantityToggle} from './QuantityToggle/QuantityToggle'
// export type { QuantityToggleProps } from './QuantityToggle/QuantityToggle'
// export { default as Ratio } from './Ratio';
// export type { RatioProps } from './Ratio';
export * from './Row' 
// export { default as Row } from '../Row';
// export type { RowProps } from '../Row';
export * from './SideNav'
// export { default as SideNav } from './SideNav/SideNav';
// export type { SideNavProps } from './SideNav/SideNav';

// export { default as Spinner } from './Spinner';
// export type { SpinnerProps } from './Spinner';


export * from './SplitButton'
// export { default as SplitButton } from '../SplitButton/SplitButton';
// export type { SplitButtonProps } from '../SplitButton/SplitButton';

// export { default as SSRProvider } from './SSRProvider';
// export type { SSRProviderProps } from './SSRProvider';

// export { default as Stack } from './Stack';
// export type { StackProps } from './Stack';
export * from './Stepper'
// export { default as Stepper } from './Stepper/Stepper'
// export { default as useStep } from './Stepper/useStep'
// export type { OnDone, StepMetadata, State as StepperState, UseStepMethods} from './Stepper/useStep'

// export { default as Tab } from './Tabs/Tab';
// export type { TabProps } from './Tabs/Tab';

// export { default as TabContainer } from './Tabs/TabContainer';
// export type { TabContainerProps } from './Tabs/TabContainer';

// export { default as TabContent } from './Tabs/TabContent';
export * from './Table'
// export { default as Table } from './Table/Table';
// export type { TableProps } from './Table/Table';

// export { default as TabPane } from './Tabs/TabPane';
// export type { TabPaneProps } from './Tabs/TabPane';

// export { default as Tabs } from './Tabs/Tabs';
// export type { TabsProps } from './Tabs/Tabs';
export * from './ThemeProvider'
// export { default as ThemeProvider } from './ThemeProvider/ThemeProvider';
// export type { ThemeProviderProps } from './ThemeProvider/ThemeProvider';
export * from './Toast'
// export { default as Toast } from './Toast/Toast';
// export type { ToastProps } from './Toast/Toast';

// export { default as ToastBody } from './Toast/ToastBody';
// export { default as ToastHeader } from './Toast/ToastHeader';
// export type { ToastHeaderProps } from './Toast/ToastHeader';

// export { default as ToastContainer } from './Toast/ToastContainer';
// export type { ToastContainerProps } from './Toast/ToastContainer';

// export { default as ToggleButton } from './ToggleButton';
// export type { ToggleButtonProps } from './ToggleButton';

// export { default as ToggleButtonGroup } from './ToggleButtonGroup';
// export type {
//   ToggleButtonCheckboxProps,
//   ToggleButtonGroupProps,
//   ToggleButtonRadioProps,
// } from './ToggleButtonGroup';
export * from './Tooltip'
// export { default as TooltipBox } from './Tooltip/TooltipBox';
// export type { TooltipBoxProps } from './Tooltip/TooltipBox';

// export { default as Tooltip } from './Tooltip/Tooltip';
// export type { TooltipProps } from './Tooltip/Tooltip';


