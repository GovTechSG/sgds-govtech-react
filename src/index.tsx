
export * from '../src/components/index'
// export * from '../src/components/Alert/Alert'
// export * from '../src/components/Button/Button'
// export * from '../src/components/Card/Card'
// export * from '../src/components/Dropdown/Dropdown'
// export * from '../src/components/Form/Form'
// export * from '../src/components/Modal/Modal'
// export * from '../src/components/Badge/Badge'
// export * from '../src/components/Breadcrumb/Breadcrumb'
// export * from '../src/components/Card/Card'
// export * from '../src/components/Card/CardImg'
// export * from '../src/components/Card/CardGroup'
// export * from '../src/components/Card/Card'
// export * from '../src/components/CloseButton/CloseButton'
// export * from '../src/components/Collapse/Collapse';
// export * from '../src/components/Dropdown/Dropdown';
// export * from '../src/components/Dropdown/DropdownButton';
// export * from '../src/components/Fade/Fade';
// export * from '../src/components/Form/Form';
// export * from '../src/components/Form/FormControl';
// export * from '../src/components/Form/FormCheck';
// export * from '../src/components/Form/FormGroup';
// export * from '../src/components/Form/FormLabel';
// export * from '../src/components/Form/FormText';
// export * from '../src/components/Form/FormSelect';
// export * from '../src/components/InputGroup/InputGroup';
// export * from '../src/components/Modal/Modal';
// export * from '../src/components/Modal/ModalBody';
// export * from '../src/components/Modal/ModalDialog';
// export * from '../src/components/Modal/ModalFooter';
// export * from '../src/components/Modal/ModalTitle';
// export * from '../src/components/Overlay/Overlay';
// export * from '../src/components/PaginationBase/PageItem';
// export * from '../src/components/PaginationBase/PaginationBase';
// export * from '../src/components/Paginations/Paginations';
// export * from '../src/components/ProgressBar/ProgressBar';
// export * from '../src/components/QuantityToggle/QuantityToggle'
// export * from '../src/components/Row';
// export * from '../src/components/SplitButton/SplitButton';
// export * from '../src/components/Stepper/Stepper'
// export * from '../src/components/Stepper/useStep'
// export * from '../src/components/Table/Table';
// export * from '../src/components/ThemeProvider/ThemeProvider';
// export * from '../src/components/Toast/Toast';
// export * from '../src/components/Toast/ToastBody';
// export * from '../src/components/Toast/ToastHeader';
// export * from '../src/components/Toast/ToastContainer';
// export * from '../src/components/Tooltip/TooltipBox';
// export * from '../src/components/Tooltip/Tooltip';