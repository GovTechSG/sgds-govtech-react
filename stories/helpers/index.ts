import { ModalProps as BaseModalProps } from '@restart/ui/Modal';

export const commonInputPropsArgType = (modalProps: BaseModalProps) => {
  const argTypes = Object.entries(modalProps).map(([key, value]) => {
    //@ts-ignore
    const defaultValue = value.default || null;
    const propType = checkPropType(value.type);
    const type = { name: propType };

    return { [key]: { type, defaultValue, table: { defaultValue } } };
  });
  const argTypesObj = Object.assign({}, ...argTypes);
  return argTypesObj;
};

const checkPropType = (type: any) => {
  if (type === String) {
    return 'string';
  }
  if (type === Boolean) return 'boolean';

  return 'object';
};

export const HIDDEN_COMMON_PROPS = {
  bsPrefix: { table: { disable: true } },
  as: { table: { disable: true } },
};

export const VARIANT = [
  'primary',
  'secondary',
  'success',
  'danger',
  'warning',
  'info',
  'dark',
  'light',
];
