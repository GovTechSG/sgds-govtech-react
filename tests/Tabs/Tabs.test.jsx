import { fireEvent, render, screen } from '@testing-library/react';
import { Tab, Tabs } from '../../src';
import React from 'react';

import { shouldWarn } from '../helpers';

const checkEventKey = (elem, eventKey) =>
  elem.getAttribute('data-rr-ui-event-key') === `${eventKey}` &&
  elem.getAttribute('id') === `test-tab-${eventKey}` &&
  elem.getAttribute('aria-controls') === `test-tabpane-${eventKey}`;

describe('<Tabs>', () => {
  it('Should show the correct tab and assign correct eventKeys', () => {
    const { getByText } = render(
      <Tabs id="test" defaultActiveKey={1}>
        <Tab title="Tab 1 title" eventKey={1}>
          Tab 1 content
        </Tab>
        <Tab title="Tab 2 title" eventKey={2}>
          Tab 2 content
        </Tab>
      </Tabs>
    );
    const firstTabButton = getByText('Tab 1 title');
    const firstTabContent = getByText('Tab 1 content');
    const secondTabButton = getByText('Tab 2 title');

    expect(firstTabButton.tagName.toLowerCase()).toEqual('button');
    expect(firstTabButton.classList).toContain('active');
    expect(firstTabContent.classList).toContain('active');

    expect(secondTabButton.classList).not.toContain('active');
    expect(secondTabButton.tagName.toLowerCase()).toEqual('button');

    expect(checkEventKey(firstTabButton, 1)).toBe(true);
    expect(checkEventKey(secondTabButton, 2)).toBe(true);
  });

  it('should get defaultActiveKey (if null) from first child tab with eventKey', () => {
    const { getByText } = render(
      <Tabs id="test" data-testid="test-id">
        <Tab title="Tab 1 title" eventKey={1}>
          Tab 1 content
        </Tab>
        <Tab title="Tab 2 title" eventKey={2}>
          Tab 2 content
        </Tab>
      </Tabs>
    );

    const firstTabButton = getByText('Tab 1 title');
    const firstTabContent = getByText('Tab 1 content');
    const secondTabButton = getByText('Tab 2 title');

    expect(firstTabButton.tagName.toLowerCase()).toEqual('button');
    expect(firstTabButton.classList).toContain('active');
    expect(firstTabContent.classList).toContain('active');

    expect(secondTabButton.classList).not.toContain('active');
    expect(secondTabButton.tagName.toLowerCase()).toEqual('button');
  });

  it('Should allow tab title to have React components', () => {
    const tabTitle = <strong className="special-tab">React Tab 2</strong>;

    const { getByText } = render(
      <Tabs id="test" defaultActiveKey={2}>
        <Tab title="Tab 1" eventKey={1}>
          Tab 1 content
        </Tab>
        <Tab title={tabTitle} eventKey={2}>
          Tab 2 content
        </Tab>
      </Tabs>
    );
    expect(getByText('React Tab 2').classList).toContain('special-tab');
  });

  it('Should call onSelect when tab is selected', () => {
    // const onSelect = (key) => {
    //   expect(key).toEqual('2');
    // };
    // const onSelectSpy = sinon.spy(onSelect);
    const onSelectSpy = jest.fn();

    const { getByText } = render(
      <Tabs id="test" onSelect={onSelectSpy} activeKey={1}>
        <Tab title="Tab 1" eventKey="1">
          Tab 1 content
        </Tab>
        <Tab title="Tab 2" eventKey="2">
          Tab 2 content
        </Tab>
      </Tabs>
    );

    fireEvent.click(getByText('Tab 2'));
    expect(onSelectSpy).toHaveBeenCalled();
  });

  it('Should have children with the correct DOM properties', () => {
    const { getByText } = render(
      <Tabs id="test" defaultActiveKey={1}>
        <Tab title="Tab 1" className="custom" eventKey={1}>
          Tab 1 content
        </Tab>
        <Tab title="Tab 2" tabClassName="tcustom" eventKey={2}>
          Tab 2 content
        </Tab>
      </Tabs>
    );
    const firstTabContent = getByText('Tab 1 content');
    const secondTabContent = getByText('Tab 2 content');

    const firstTabTitle = getByText('Tab 1');
    const secondTabTitle = getByText('Tab 2');

    expect(firstTabContent.classList).toContain('custom');
    expect(secondTabContent.classList).not.toContain('tcustom');

    expect(firstTabTitle.classList).not.toContain('custom');
    expect(secondTabTitle.classList).toContain('tcustom');
  });

  it('Should have variant named tabs-basic-toggle', () => {
    const { getByTestId } = render(
      <Tabs
        data-testid="test"
        variant="tabs-basic-toggle"
        defaultActiveKey={1}
        transition={false}
      >
        <Tab title="Tab 1" eventKey={1}>
          Tab 1 content
        </Tab>
        <Tab title="Tab 2" eventKey={2}>
          Tab 2 content
        </Tab>
      </Tabs>
    );

    expect(getByTestId('test')).toHaveAttribute('variant', 'tabs-basic-toggle');
  });

  it('Should have variant named tabs-info-toggle', () => {
    const { getByTestId } = render(
      <Tabs
        data-testid="test"
        variant="tabs-info-toggle"
        defaultActiveKey={1}
        transition={false}
      >
        <Tab title="Tab 1" eventKey={1}>
          Tab 1 content
        </Tab>
        <Tab title="Tab 2" eventKey={2}>
          Tab 2 content
        </Tab>
      </Tabs>
    );

    expect(getByTestId('test')).toHaveAttribute('variant', 'tabs-info-toggle');
  });

  it('it should display icon for contentLeft prop before title for tabs-basic-toggle', () => {
    const { container } = render(
      <Tabs
        data-testid="test"
        variant="tabs-basic-toggle"
        defaultActiveKey={1}
        transition={false}
      >
        <Tab
          title="Tab 1"
          eventKey={1}
          contentLeft={<i className="bi bi-house left"></i>}
        >
          Tab 1 content
        </Tab>
        <Tab title="Tab 2" eventKey={2}>
          Tab 2 content
        </Tab>
      </Tabs>
    );

    expect(
      container.querySelector('button>i.bi.bi-house.left')
    ).toBeInTheDocument();
  });

  it('it should only render class `has-icon` when contentLeft prop is being used for tabs-info-toggle', () => {
    const { container } = render(
      <Tabs
        data-testid="test"
        variant="tabs-info-toggle"
        defaultActiveKey={1}
        transition={false}
      >
        <Tab
          title="Tab 1"
          eventKey={1}
          contentLeft={<i className="bi bi-house left"></i>}
        >
          Tab 1 content
        </Tab>
        <Tab title="Tab 2" eventKey={2}>
          Tab 2 content
        </Tab>
      </Tabs>
    );

    expect(
      container.querySelector(
        'button>div.tabs-info-label.has-icon>i.bi.bi-house.left'
      )
    ).toBeInTheDocument();
  });

  it('it should display icon for contentRight prop after title for tabs-basic-toggle', () => {
    const { container } = render(
      <Tabs
        data-testid="test"
        variant="tabs-basic-toggle"
        defaultActiveKey={1}
        transition={false}
      >
        <Tab
          title="Tab 1"
          eventKey={1}
          contentRight={<i className="bi bi-house left"></i>}
        >
          Tab 1 content
        </Tab>
        <Tab title="Tab 2" eventKey={2}>
          Tab 2 content
        </Tab>
      </Tabs>
    );

    expect(
      container.querySelector('button>i.bi.bi-house.left')
    ).toBeInTheDocument();
  });

  it('it should display 99 for contentBottom prop below title for tabs-info-toggle', () => {
    const { container } = render(
      <Tabs variant="tabs-info-toggle" defaultActiveKey={1} transition={false}>
        <Tab title="Tab 1" eventKey={1} contentBottom={99}>
          Tab 1 content
        </Tab>
        <Tab title="Tab 2" eventKey={2}>
          Tab 2 content
        </Tab>
      </Tabs>
    );
    const divContentBottom = container.querySelector(
      'button>div.tabs-info-count'
    );
    expect(divContentBottom.textContent).toBe('99');
  });

  it('it should NOT display icon for contentRight prop after title for tabs-info-toggle', () => {
    const { asFragment, container } = render(
      <Tabs
        data-testid="test"
        variant="tabs-info-toggle"
        defaultActiveKey={1}
        transition={false}
      >
        <Tab
          title="Tab 1"
          eventKey={1}
          contentRight={<i className="bi bi-house left"></i>}
        >
          Tab 1 content
        </Tab>
        <Tab title="Tab 2" eventKey={2}>
          Tab 2 content
        </Tab>
      </Tabs>
    );

    expect(
      container.querySelector('button>i.bi.bi-house.left')
    ).not.toBeInTheDocument();
  });

  it('it should NOT display icon for contentBottom prop after title for tabs-basic-toggle', () => {
    const { container } = render(
      <Tabs
        data-testid="test"
        variant="tabs-basic-toggle"
        defaultActiveKey={1}
        transition={false}
      >
        <Tab
          title="Tab 1"
          eventKey={1}
          contentBottom={<i className="bi bi-house left"></i>}
        >
          Tab 1 content
        </Tab>
        <Tab title="Tab 2" eventKey={2}>
          Tab 2 content
        </Tab>
      </Tabs>
    );

    expect(
      container.querySelector('button>i.bi.bi-house.left')
    ).not.toBeInTheDocument();
  });

  it('Should pass disabled to Nav', () => {
    const onSelect = (e) => e;
    const onSelectSpy = jest.fn();

    const { getByText } = render(
      <Tabs id="test" defaultActiveKey={1} onSelect={onSelectSpy}>
        <Tab title="Tab 1" eventKey={1}>
          Tab 1 content
        </Tab>
        <Tab title="Tab 2" eventKey={2} disabled>
          Tab 2 content
        </Tab>
      </Tabs>
    );
    const secondTabTitle = getByText('Tab 2');
    expect(secondTabTitle.classList).toContain('disabled');

    expect(onSelectSpy).not.toHaveBeenCalled();
  });

  it('Should not render a Tab without a title', () => {
    shouldWarn('Failed prop');
    const { getByTestId, asFragment } = render(
      <Tabs data-testid="testid" id="test" defaultActiveKey={1}>
        {/* @ts-ignore */}
        <Tab eventKey={1}>Tab 1 content</Tab>
        <Tab title="Tab 2" eventKey={2} disabled>
          Tab 2 content
        </Tab>
      </Tabs>
    );
    expect(asFragment()).toMatchSnapshot();
    const tabs = getByTestId('testid');
    expect(tabs.children.length).toEqual(1);
  });

  it('Should render TabPane with role="tab"', () => {
    const { getAllByRole, asFragment } = render(
      <Tabs data-testid="testid" id="test" defaultActiveKey={1}>
        <Tab title="Tab 1" eventKey={1}>
          Tab 1 content
        </Tab>
      </Tabs>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(getAllByRole('tab').length).toEqual(1);
  });

  it('should have fade animation by default', () => {
    render(
      <Tabs id="test" defaultActiveKey={1}>
        <Tab title="Tab 1" eventKey={1}>
          tab 1 content
        </Tab>
      </Tabs>
    );
    const input = screen.getByText(/tab 1 content/i);

    expect(input.classList).toContain('fade');
  });

  it('Should omit Transition in TabPane if prop is false ', () => {
    const { getByText } = render(
      <Tabs id="test" defaultActiveKey={1}>
        <Tab title="Tab 1" className="custom" eventKey={1} transition={false}>
          Tab 1 content
        </Tab>
        <Tab title="Tab 2" tabClassName="tcustom" eventKey={2}>
          Tab 2 content
        </Tab>
      </Tabs>
    );
    const firstTabContent = getByText('Tab 1 content');
    const secondTabContent = getByText('Tab 2 content');

    expect(firstTabContent.classList).not.toContain('fade');
    expect(secondTabContent.classList).toContain('fade');
  });
});

// describe('<Tab>', () => {
//   it('should throw error message on attempt to render', () => {
//     expect(() =>
//       render(
//         <Tab title="Tab 1" eventKey={1}>
//           Tab 1 content
//         </Tab>,
//       ),
//     ).to.throw();
//   });
// });
