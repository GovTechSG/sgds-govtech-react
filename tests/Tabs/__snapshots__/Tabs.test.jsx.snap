// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs> Should not render a Tab without a title 1`] = `
<DocumentFragment>
  <ul
    class="nav sgds nav-tabs"
    data-testid="testid"
    role="tablist"
  >
    <li
      class="nav-item"
      role="presentation"
    >
      <button
        aria-controls="test-tabpane-2"
        aria-disabled="true"
        class="nav-link disabled"
        data-rr-ui-event-key="2"
        disabled=""
        id="test-tab-2"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Tab 2
      </button>
    </li>
  </ul>
  <div
    class="sgds tab-content"
  >
    <div
      aria-labelledby="test-tab-1"
      class="fade tab-pane active show"
      id="test-tabpane-1"
      role="tabpanel"
    >
      Tab 1 content
    </div>
    <div
      aria-labelledby="test-tab-2"
      class="fade tab-pane"
      id="test-tabpane-2"
      role="tabpanel"
    >
      Tab 2 content
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Tabs> Should render TabPane with role="tab" 1`] = `
<DocumentFragment>
  <ul
    class="nav sgds nav-tabs"
    data-testid="testid"
    role="tablist"
  >
    <li
      class="nav-item"
      role="presentation"
    >
      <button
        aria-controls="test-tabpane-1"
        aria-selected="true"
        class="nav-link active"
        data-rr-ui-event-key="1"
        id="test-tab-1"
        role="tab"
        type="button"
      >
        Tab 1
      </button>
    </li>
  </ul>
  <div
    class="sgds tab-content"
  >
    <div
      aria-labelledby="test-tab-1"
      class="fade tab-pane active show"
      id="test-tabpane-1"
      role="tabpanel"
    >
      Tab 1 content
    </div>
  </div>
</DocumentFragment>
`;
